### PR TITLE
feat: loading screens

### PR DESCRIPTION
**Pull request by tataihono**
# Description

No more SSR for journey editor. Instead, we'll client-side load everything and put skeletons in place to make it feel more responsive. Everything here should be tested via VR testing since they are purely visual changes.

The bigger part of this PR is that journey is now nullable in journey provider and steps is now nullable in editor provider. These two changes alone make for significant null check additions across the app.

- [Link to Basecamp Todo](https://3.basecamp.com/3105655/buckets/26787371/todos/4757422949)

# How should this PR be QA Tested?

Please describe the QA tests that you ran to verify your changes. Please also note any relevant details for your test configuration.

- [ ] load journeys list
- [ ] load journey view
- [ ] load journey edit

# Checklist

- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [x] Any dependent changes have been merged into main

### CODE REVIEW

File: apps/journeys-admin/src/components/JourneyView/Properties/Properties.tsx [hunk 1]
```diff
@@ -35,24 +35,28 @@ export function Properties(): ReactElement {
           <Box sx={{ px: 6 }}>
             <JourneyDetails />
           </Box>
-          {userJourneys != null && (
```
Comment (csiyang) [inline apps/journeys-admin/src/components/JourneyView/Properties/Properties.tsx:38]: why isn't this check needed anymore?

File: apps/journeys-admin/pages/index.tsx [hunk 2]
```diff
@@ -46,25 +45,18 @@ function IndexPage(): ReactElement {
...
-        {data?.journeys != null && <JourneyList journeys={data.journeys} />}
+        <JourneyList journeys={data?.journeys} />
       </PageWrapper>
     </>
   )
 }
 
 export const getServerSideProps = withAuthUserTokenSSR({
   whenUnauthed: AuthAction.REDIRECT_TO_LOGIN
-})(async ({ AuthUser }) => {
```
Comment (csiyang) [inline apps/journeys-admin/pages/index.tsx:57]: Where did this authentication stuff get moved too? Couldn't find it, or do we not need it anymore? 
Same on [journeySlug] and [journeySlug]/edit

Comment (tataihono) [reply apps/journeys-admin/pages/index.tsx:57]: still runs authentication in the withAuthUserTokenSSR method. We just don't use it server-side anymore.

File: apps/journeys-admin/src/components/JourneyView/Properties/Properties.tsx (hunk 1 above)
Comment (tataihono) [reply apps/journeys-admin/src/components/JourneyView/Properties/Properties.tsx:38]: because the underlying AccessAvatars handles the null case.

**Review by csiyang — APPROVED** [review]

**Pull request merged**